src/Framework/FeatureAvailabilityChecker.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Build.Framework
+{
+    /// <summary>
+    /// The status of a feature.
+    /// </summary>
+    public enum FeatureStatus
+    {
+        /// <summary>
+        /// The feature is not found.
+        /// </summary>
+        NotFound,
+
+        /// <summary>
+        /// The feature is available.
+        /// </summary>
+        Available,
```

Comment:
I think we might need `Preview` status so clients can choose if it will use feature if it is stable vs preview.
@baronfel do you agree?

Reply:
That seems useful, yeah.

Reply:
Done